core/src/main/java/uk/gov/gchq/koryphe/impl/function/If.java

```diff
@@ -81,14 +82,7 @@
      */
     @Override
     public O apply(final I input) {
-        boolean conditionTmp;
-        if (null == condition) {
-            conditionTmp = null != predicate && predicate.test(input);
-        } else {
-            conditionTmp = condition;
-        }
-
-        if (conditionTmp) {
+        if (Objects.requireNonNullElseGet(condition, () -> null != predicate && predicate.test(input))) {
```

Comment:
👍 
 `Returns the first argument if it is non-null and otherwise returns the non-null value of supplier.get().`

Reply:
Question why is condition `Boolean` and not `boolean` ? Could that remove the need to null check?
I understand and happy that refactoring some logic is out of scope for this ticket.

Reply:
I think that this is the case for most of the functions which take in boolean inputs - is this to do with the output of previous actions potentially being `null` and us wanting to explicitly define the behaviour when the input to these function is `null`?